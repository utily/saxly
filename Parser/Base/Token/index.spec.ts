import "jest"
import { Token } from "./"

async function tokenize(data: string): Promise<Token[]> {
	const result: Token[] = []
	for await (const token of Token.tokenize(data))
		result.push(token)
	return result
}

describe("saxly.Parser.Base.Token", () => {
	it("parse <root/>", async () => expect(await tokenize("<root/>")).toMatchSnapshot())
	it("parse <root></root>", async () => expect(await tokenize("<root></root>")).toMatchSnapshot())
})
