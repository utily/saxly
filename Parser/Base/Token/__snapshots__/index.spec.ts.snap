// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`saxly.Parser.Base.Token parse <root/> 1`] = `
Array [
  Token {
    "content": "<",
    "range": Object {
      "end": Object {
        "column": 1,
        "row": 0,
      },
      "start": Object {
        "column": 0,
        "row": 0,
      },
    },
    "type": "symbol",
  },
  Token {
    "content": "root",
    "range": Object {
      "end": Object {
        "column": 2,
        "row": 0,
      },
      "start": Object {
        "column": 1,
        "row": 0,
      },
    },
    "type": "text",
  },
  Token {
    "content": "/>",
    "range": Object {
      "end": Object {
        "column": 3,
        "row": 0,
      },
      "start": Object {
        "column": 2,
        "row": 0,
      },
    },
    "type": "symbol",
  },
]
`;

exports[`saxly.Parser.Base.Token parse <root></root> 1`] = `
Array [
  Token {
    "content": "<",
    "range": Object {
      "end": Object {
        "column": 1,
        "row": 0,
      },
      "start": Object {
        "column": 0,
        "row": 0,
      },
    },
    "type": "symbol",
  },
  Token {
    "content": "root",
    "range": Object {
      "end": Object {
        "column": 2,
        "row": 0,
      },
      "start": Object {
        "column": 1,
        "row": 0,
      },
    },
    "type": "text",
  },
  Token {
    "content": ">",
    "range": Object {
      "end": Object {
        "column": 3,
        "row": 0,
      },
      "start": Object {
        "column": 2,
        "row": 0,
      },
    },
    "type": "symbol",
  },
  Token {
    "content": "</",
    "range": Object {
      "end": Object {
        "column": 4,
        "row": 0,
      },
      "start": Object {
        "column": 3,
        "row": 0,
      },
    },
    "type": "symbol",
  },
  Token {
    "content": "root",
    "range": Object {
      "end": Object {
        "column": 5,
        "row": 0,
      },
      "start": Object {
        "column": 4,
        "row": 0,
      },
    },
    "type": "text",
  },
  Token {
    "content": ">",
    "range": Object {
      "end": Object {
        "column": 6,
        "row": 0,
      },
      "start": Object {
        "column": 5,
        "row": 0,
      },
    },
    "type": "symbol",
  },
]
`;
